zerver/lib/actions.py

```diff
@@ -3614,6 +3647,18 @@ def get_non_subscribed_subs() -> List[Tuple[UserProfile, Stream]]:
         streams_by_user[sub_info.user.id].append(stream)
         altered_user_dict[stream.id].add(sub_info.user.id)
 
+    for sub in removed_stream_admin_list:
+        event = dict(
+            type="subscription",
+            op="update",
+            email=sub.user.email,
+            property="role",
+            value=Subscription.ROLE_MEMBER,
+            stream_id=sub.stream.id,
+            name=sub.stream.name,
+        )
```

Comment:
This should definitely call a `notify_change_stream_role` function that will be shared with the code path for changing stream-level roles; otherwise we're going to end up with bugs where we update one of the copies of this event format and not others.

Also, we should not send email/stream_name fields (we want to work only with `id`s instead) and actually send a list of user_ids and stream_ids, so we can use a single bulk event where convenient, even if we're doing a loop for simplicity of implementation, so that we don't need to change the protocol to later optimize this.  (We had real performance issues with this type of code path in the past, though with more common operations).

Reply:
@timabbott I agree with sending only ids in event, should we also change the event dict sent from `do_change_subscription_property` to only send id and not user email or stream name. I am suggesting this because event sent from `do_change_subscription_role` and `do_change_subscription_property` have same event `type=subscription` and `op=update` and the openapi schema is not validating in tests because both functions send different details in event objects. There is another way by sending the different event type for role case, but I think we would eventually prefer only ids and not stream name and user email.